bug(browser): Playwright Chromium browser fails to launch after service starts

Playwright service starts successfully and listens on port 9001, but fails to initialize Chromium browser during pre-initialization.

## Symptoms
- Service running: ✓
- Port 9001 listening: ✓
- Browser launch: ✗ FAILS
- Error: 'Failed to launch browser'

## Timeline
1. Service starts successfully (express module loaded)
2. Port 9001 bound successfully
3. Browser pre-initialization triggered
4. Chromium launch fails with generic error

## Logs
```
2026-04-08T21:20:25.831Z [ERROR] Failed to launch browser: {"name":"Error"}
2026-04-08T21:20:25.831Z [ERROR] Failed to pre-initialize browser: {"name":"Error"}
```

## Investigation Needed
- Check if Chromium is installed on host
- Check DISPLAY variable (currently :1)
- Check VNC/X11 availability
- Review playwright error details (error object is empty)

## Environment
- Node.js v20.20.2
- Playwright 1.54.2
- HEADLESS_MODE=False
- DISPLAY=:1
- Host: WSL2 Linux

## Related
- This is separate from #4052 (port conflict - now resolved)
- Service infrastructure is correct, browser initialization is the issue

--- a/autobot-browser-worker/playwright-server.js
+++ b/autobot-browser-worker/playwright-server.js
@@ -63,7 +63,10 @@ async function initBrowser() {
     }
     return browser;
   } catch (error) {
-    logger.error('Failed to launch browser:', error);
+    logger.error('Failed to launch browser:', error.message || error);
+    if (error.stack) {
+      logger.error('Browser launch stack:', error.stack);
+    }
     throw error;
   }
 }
@@ -825,108 +828,6 @@ app.post('/test-frontend', async (req, res) => {
       error: error.message,
       timestamp: new Date().toISOString()
     });
-  }
-});
-
-// =============================================================================
-// Persistent navigation page for BrowserTool (Issue #1120)
-// =============================================================================
-
-navPage = null;
-
-async function ensureNavPage() {
-  const b = await initBrowser();
-  if (!navPage || navPage.isClosed()) {
-    navPage = await b.newPage();
-    await navPage.setExtraHTTPHeaders({
-      'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.0.0 Safari/537.36'
-    });
-    logger.info('Persistent navigation page created');
-  }
-  return navPage;
-}
-
-async function captureNavScreenshot() {
-  const buf = await navPage.screenshot({ type: 'png' });
-  return buf.toString('base64');
-}
-
-app.get('/status', (req, res) => {
-  res.json({
-    status: (browser && browser.isConnected()) ? 'connected' : 'disconnected',
-    timestamp: new Date().toISOString(),
-    browser_connected: browser ? browser.isConnected() : false,
-    page_open: navPage !== null && !navPage.isClosed()
-  });
-});
-
-app.post('/navigate', async (req, res) => {
-  const { url } = req.body;
-  if (!url) {
-    return res.status(400).json({ success: false, error: 'url is required' });
-  }
-  logger.info('Navigate request:', url);
-  try {
-    const p = await ensureNavPage();
-    if (url === 'javascript:history.back()') {
-      await p.goBack({ waitUntil: 'domcontentloaded', timeout: 15000 }).catch(() => {});
-    } else if (url === 'javascript:history.forward()') {
-      await p.goForward({ waitUntil: 'domcontentloaded', timeout: 15000 }).catch(() => {});
-    } else if (url === 'javascript:location.reload()') {
-      await p.reload({ waitUntil: 'domcontentloaded', timeout: 15000 });
-    } else {
-      await p.goto(url, { waitUntil: 'domcontentloaded', timeout: 30000 });
-    }
-    res.json({ success: true, url: p.url(), title: await p.title() });
-  } catch (error) {
-    logger.error('Navigate error:', error);
-    res.status(500).json({ success: false, error: error.message });
-  }
-});
-
-app.post('/screenshot', async (req, res) => {
-  logger.info('Screenshot request');
-  try {
-    const p = await ensureNavPage();
-    const buf = await p.screenshot({ type: 'png', fullPage: false });
-    const b64 = buf.toString('base64');
-    res.json({ success: true, screenshot: b64, timestamp: new Date().toISOString() });
-  } catch (error) {
-    logger.error('Screenshot error:', error);
-    res.status(500).json({ success: false, error: error.message });
-  }
-});
-
-app.post('/back', async (req, res) => {
-  try {
-    const p = await ensureNavPage();
-    await p.goBack({ waitUntil: 'domcontentloaded', timeout: 15000 }).catch(() => {});
-    res.json({ success: true, url: p.url(), title: await p.title() });
-  } catch (error) {
-    logger.error('Back error:', error);
-    res.status(500).json({ success: false, error: error.message });
-  }
-});
-
-app.post('/forward', async (req, res) => {
-  try {
-    const p = await ensureNavPage();
-    await p.goForward({ waitUntil: 'domcontentloaded', timeout: 15000 }).catch(() => {});
-    res.json({ success: true, url: p.url(), title: await p.title() });
-  } catch (error) {
-    logger.error('Forward error:', error);
-    res.status(500).json({ success: false, error: error.message });
-  }
-});
-
-app.post('/reload', async (req, res) => {
-  try {
-    const p = await ensureNavPage();
-    await p.reload({ waitUntil: 'domcontentloaded', timeout: 15000 });
-    res.json({ success: true, url: p.url(), title: await p.title() });
-  } catch (error) {
-    logger.error('Reload error:', error);
-    res.status(500).json({ success: false, error: error.message });
   }
 });
 

--- a/autobot-infrastructure/autobot-browser-worker/templates/autobot-browser-worker.service
+++ b/autobot-infrastructure/autobot-browser-worker/templates/autobot-browser-worker.service
@@ -1,6 +1,6 @@
 # AutoBot Browser Worker - Systemd Service Unit
 # Deploys to: 172.16.168.25 (Browser VM)
-# Runs Playwright browser automation service
+# Runs Playwright browser automation service via Node.js/Express
 #
 # Installation:
 #   sudo cp autobot-browser-worker.service /etc/systemd/system/
@@ -20,39 +20,31 @@ User=autobot
 Group=autobot
 WorkingDirectory=/opt/autobot/autobot-browser-worker
 
-# Environment
-Environment="PYTHONUNBUFFERED=1"
-Environment="PYTHONDONTWRITEBYTECODE=1"
+# Environment variables
+EnvironmentFile=/etc/autobot/autobot-browser-worker.env
+Environment="NODE_ENV=production"
 Environment="PLAYWRIGHT_BROWSERS_PATH=/opt/autobot/autobot-browser-worker/browsers"
-EnvironmentFile=-/opt/autobot/autobot-browser-worker/.env
 
-# Start uvicorn server
-ExecStart=/opt/autobot/autobot-browser-worker/venv/bin/uvicorn \
-    main:app \
-    --host 0.0.0.0 \
-    --port 3000 \
-    --workers 1 \
-    --loop uvloop \
-    --log-level info
+# Start Node.js Express server
+ExecStart=/usr/bin/node /opt/autobot/autobot-browser-worker/playwright-server.js
 
 # Restart policy
-Restart=on-failure
-RestartSec=10
-TimeoutStartSec=60
+Restart=always
+RestartSec=5s
+TimeoutStartSec=30
 TimeoutStopSec=30
 
 # Resource limits (browser automation needs memory)
 MemoryMax=4G
 CPUQuota=150%
+LimitNOFILE=65535
 
 # Security
 PrivateTmp=true
 NoNewPrivileges=true
 ProtectSystem=strict
-ReadWritePaths=/opt/autobot/autobot-browser-worker/data
-ReadWritePaths=/opt/autobot/autobot-browser-worker/screenshots
-ReadWritePaths=/opt/autobot/autobot-browser-worker/logs
 ReadWritePaths=/opt/autobot/autobot-browser-worker/browsers
+ReadWritePaths=/var/log/autobot
 
 # Logging
 StandardOutput=journal

--- a/autobot-slm-backend/ansible/roles/browser/templates/playwright.service.j2
+++ b/autobot-slm-backend/ansible/roles/browser/templates/playwright.service.j2
@@ -12,7 +12,7 @@ WorkingDirectory=/opt/autobot/autobot-browser-worker
 # Environment
 Environment="PLAYWRIGHT_PORT={{ playwright_port }}"
 Environment="BROWSER_TIMEOUT={{ browser_timeout }}"
-Environment="HEADLESS_MODE={{ headless_mode }}"
+Environment="HEADLESS={{ headless_mode | lower }}"
 Environment="DISPLAY={{ vnc_display }}"
 
 # Start command - Node.js Express server

